### PR TITLE
[VP9][Encode] Fix unaligned height static content encoding quality

### PR DESCRIPTION
**Pull request by daijh**
If the padding row is N, repeatly copy the last row into the N padding rows, instead of copy whole last N rows into padding.

### CODE REVIEW

**Review by XinfengZhang — APPROVED** [review]

File: media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp [hunk 1]
```diff
@@ -3493,29 +3493,23 @@ void CodechalVdencVp9StateG12::fill_pad_with_value(PMOS_SURFACE psSurface, uint3
...
-
-        if (src_data_uv_end - y_pad_length > src_data_y_end)
-        {
-            memcpy_s(src_data_uv_end, y_pad_length, src_data_uv_end - y_pad_length, y_pad_length);
-        }    
+        uint8_t *src_data_uv     = src_data + UVPlaneOffset;
+        uint32_t uv_plane_size   = pitch * real_height / 2;
+        uint8_t *src_data_uv_end = src_data_uv + uv_plane_size;
+        uint32_t uv_pitch = pitch / 2;
+        for (int i = 0; i < pad_rows; i++) {
```
Comment (IlyaSergeyev) [inline media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp:3510]: Why you do not use uint32_t for i here?
Signed int i causes warning in comparition and error if warnings are treated as errors.

**Review by IlyaSergeyev — CHANGES_REQUESTED** [review]

File: media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp [hunk 2]
```diff
@@ -3493,29 +3493,23 @@ void CodechalVdencVp9StateG12::fill_pad_with_value(PMOS_SURFACE psSurface, uint3
...
-
-        if (src_data_y_end > src_data_y_end - y_pad_length)
-        {
-            memcpy_s(src_data_y_end, y_pad_length, src_data_y_end - y_pad_length, y_pad_length);
+        uint8_t *src_data_y     = src_data + YPlaneOffset;
+        uint32_t y_plane_size   = pitch * real_height;
+        uint8_t *src_data_y_end = src_data_y + y_plane_size;
+        uint32_t y_pitch = pitch;
+        for (int i = 0; i < pad_rows; i++) {
+          memcpy_s(src_data_y_end + i * y_pitch, y_pitch, src_data_y_end - y_pitch, y_pitch);
```
Comment (IlyaSergeyev) [inline media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp:3503]: MOS_SecureMemcpy may be better here

**Review by IlyaSergeyev — CHANGES_REQUESTED** [review]

File: media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp (hunk 1 above)
Comment (daijh) [reply media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp]: Good suggestion.

File: media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp (hunk 2 above)
Comment (daijh) [reply media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp]: Done

**Review by IlyaSergeyev — APPROVED** [review]

**Pull request merged**